src/controller/actions/topicDashboard.js

```diff
@@ -0,0 +1,113 @@
+import api from '../../controller/api';
+
+export const RESOURCE_LOADING = 'RESOURCE_LOADING';
+export const resourceLoading = () => ({
+  type: RESOURCE_LOADING
+});
+
+export const RESOURCE_SUCCESS = 'RESOURCE_SUCCESS';
+export const resourceSuccess = data => ({
+  type: RESOURCE_SUCCESS,
+  payload: data
+});
+
+export const RESOURCE_ERROR = 'RESOURCE_ERROR';
+export const resourceError = error => ({
+  type: RESOURCE_ERROR,
+  payload: error
+});
+
+export const ADD_RESOURCE = 'ADD_RESOURCE';
+export const addResource = resource => ({
+  type: ADD_RESOURCE,
+  resource
+});
+
+export const UPDATE_RESOURCE = 'UPDATE_RESOURCE';
+export const updateResource = (resource, id) => ({
+  type: UPDATE_RESOURCE,
+  resource,
+  id
+});
+
+export const DELETE_RESOURCE = 'DELETE_RESOURCES';
+export const delResource = id => ({
+  type: DELETE_RESOURCE,
+  id
+});
+
+/**
+ *Gets all resources belonging to a parent
+ *First dispatches a loading function to change state to loading
+ *Second: Sends GET request to the server using api.resources
+ * Third: Dispatches resource_success action to add new data from server to current state
+ * If there is an error, it dispatches an error obj to state
+ * *@param {{id:integer}}} id
+ */
+export const getResources = id => dispatch => {
```

Comment:
Perhaps renaming this would be beneficial. You're not really getting resources here, you're getting a topic. To coincide with the other reducers, how about `initializeTopicDashboard`?